src/base/settingsstorage.h

```diff
@@ -62,4 +62,27 @@ public slots:
     mutable QReadWriteLock m_lock;
 };
 
+template<typename T>
+class StorableValue
```

Comment:
>So this suggestion won't work too.

Why? `m_storeDialogSize(SETTINGS_KEY("Dimension"))` uses default constructed QSize which is invalid.